tftrt/examples/cpp/image-classification/main.cc

```diff
@@ -0,0 +1,168 @@
+  }
+
+  // Convert the first image to a tensorflow::Tensor.
+  tensorflow::Tensor input_image = mnist::MNISTImageToTensor(images, 1);
+  std::cout << "Created input tensor with shape "
+            << input_image.shape().DebugString() << std::endl;
+  tensorflow::Tensor input_image2 = mnist::MNISTImageToTensor(images, 4);
+  std::cout << "Created input tensor with shape "
+            << input_image2.shape().DebugString() << std::endl;
+
+  inputs->push_back({input_image});
+  inputs->push_back({input_image2});
+
+  for (int i = 0; i < 4; i++) {
+    std::cout << "Input image " << i << std::endl;
+    MNISTPrint(images[i]);
+  }
+  return tensorflow::Status::OK();
+}
+
+int main(int argc, char **argv) {
+  // Load the saved model from the provided path.
+  // std::string export_dir =
+  // "/data/tensorflow-source/tf_trt_cpp_example/mnist_model";
+  std::string export_dir =
+      "/data/tensorflow-source/tf_trt_cpp_example/mnist_model_frozen";
+  tensorflow::SavedModelBundle bundle;
+  tensorflow::Status status = LoadModel(export_dir, &bundle);
+  if (!status.ok()) {
+    std::cerr << "Error loading the model " << status << std::endl;
+    return 1;
+  }
+
+  // Prepare input tensors
+  std::vector<std::vector<tensorflow::Tensor>> inputs;
+  status = LoadInputs(&inputs);
+  if (!status.ok()) {
+    std::cerr << "Error reading input data " << status << std::endl;
+    return 1;
+  }
+
+  // Run TF-TRT conversion
+  tensorflow::tensorrt::TfTrtConversionParams params;
+  params.profile_strategy = tensorflow::tensorrt::ProfileStrategy::kOptimal;
+  tensorflow::StatusOr<tensorflow::GraphDef> status_or_gdef =
+      tensorflow::tensorrt::ConvertAndBuild(
+          bundle.meta_graph_def.graph_def(),
+          bundle.meta_graph_def.signature_def().at("serving_default"), params,
+          inputs);
+  if (!status_or_gdef.ok()) {
+    std::cerr << "Error converting the graph" << status << std::endl;
+    return 1;
+  }
+  tensorflow::GraphDef &converted_graph_def = status_or_gdef.ValueOrDie();
+  tensorflow::Session *session = nullptr;
+  status = NewSession(tensorflow::SessionOptions(), &session);
+  bundle.session.reset(session);
+  status = bundle.session->Create(converted_graph_def);
+
+  if (!status.ok()) {
+    std::cerr << "Error resetting session to converted graph " << status
+              << std::endl;
+    return 1;
+  }
+
+  // Infer the converted model
+  std::vector<std::string> input_names = {"serving_default_flatten_input"};
+  std::vector<std::string> output_names = {"PartitionedCall"};
+  for (auto const &input : inputs) {
+    std::vector<std::pair<std::string, tensorflow::Tensor>> input_pairs;
+    for (int i = 0; i < input_names.size(); i++) {
+      input_pairs.push_back({input_names.at(i), input.at(i)});
+    }
+    std::vector<tensorflow::Tensor> output_tensors;
+    for (int i = 0; i < output_names.size(); i++) {
+      output_tensors.push_back({});
+    }
+    std::cout << "Inferring the model\n";
```

Comment:
Nit: if run fails, might never see the "inferring the model" message without `std::endl`